packages/wallets/src/components/Base/WalletClipboard/__tests__/WalletClipboard.spec.tsx

```diff
@@ -0,0 +1,102 @@
+
+    beforeEach(() => {
+        mockCopy = jest.fn();
+        (useCopyToClipboard as jest.Mock).mockReturnValue([null, mockCopy]);
+        (useHover as jest.Mock).mockReturnValue(false);
+        mockUseDevice.mockReturnValue({ isMobile: false });
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('should render the button with copy icon initially', () => {
+        render(<WalletClipboard textCopy='Sample text to copy' />);
+        const button = screen.getByRole('button');
+
+        expect(button).toBeInTheDocument();
+        expect(screen.getByTestId('dt_legacy_copy_icon')).toBeInTheDocument();
+    });
+    it('should render the button with won icon when copied', async () => {
+        render(<WalletClipboard textCopy='Sample text to copy' />);
+        const button = screen.getByRole('button');
+        await userEvent.click(button);
+
+        expect(screen.getByTestId('dt_legacy_won_icon')).toBeInTheDocument();
+    });
+    it('should call copy function with textCopy when button is clicked', async () => {
+        render(<WalletClipboard textCopy='Sample text to copy' />);
+        const button = screen.getByRole('button');
+        await userEvent.click(button);
+
+        expect(mockCopy).toHaveBeenCalledWith('Sample text to copy');
+    });
+    it('should show tooltip with "Copy" message when hovered', () => {
+        (useHover as jest.Mock).mockReturnValue(true);
+        render(<WalletClipboard textCopy='Sample text to copy' />);
+
+        expect(screen.getByText('Copy')).toBeInTheDocument();
+    });
+    it('should show tooltip with "Copied!" message when clicked and hovered', async () => {
+        (useHover as jest.Mock).mockReturnValue(true);
+        render(<WalletClipboard textCopy='Sample text to copy' />);
+        const button = screen.getByRole('button');
+        await userEvent.click(button);
+
+        expect(screen.getByText('Copied!')).toBeInTheDocument();
+    });
+    it('should clear timeout on unmount', async () => {
+        jest.useFakeTimers();
+        const { unmount } = render(<WalletClipboard textCopy='Sample text to copy' />);
+        await userEvent.click(screen.getByRole('button'));
+        unmount();
+
+        jest.runAllTimers();
+        expect(clearTimeout).toHaveBeenCalled();
+    });
+    it('should not show tooltip on mobile', async () => {
+        mockUseDevice.mockReturnValue({ isMobile: true });
+        render(<WalletClipboard textCopy='Sample text to copy' />);
+        const button = screen.getByRole('button');
+        await userEvent.click(button);
+
+        expect(screen.queryByText('Copy')).not.toBeInTheDocument();
+        expect(screen.queryByText('Copied!')).not.toBeInTheDocument();
+    });
+    it('should reset the icon and message after 2 seconds', async () => {
+        jest.useFakeTimers();
+        (useHover as jest.Mock).mockReturnValue(true);
+        render(<WalletClipboard textCopy='Sample text to copy' />);
```

Comment:
It looks like you're rendering the component in the same way in each spec case.
Think about extracting this logic to custom function in the beginning of the file like:
```suggestion
        const renderComponent = (textCopy='Sample text to copy') => 
            render(<WalletClipboard textCopy={textCopy})
        
        ...
        
        // Example of usage (a little bit cleaner I suppose)
        renderComponent()
```